magento/magento2
# Improve template minifier

### Description (*)

This PR aims to improve the template minifier by :
- fixing a variety of cases that are currently handled incorrectly,
- commenting on each step of the minification process to explain what it does,
- increasing the exhaustiveness of the corresponding test.

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#32153

### Manual testing scenarios (*)

1. See magento/magento2#32153 or the [updated test](https://github.com/magento/magento2/blob/883c037bd6bad4728c092cb340f672cdc2d784a4/lib/internal/Magento/Framework/View/Test/Unit/Template/Html/MinifierTest.php#L121)

### Questions or comments

It is very likely that there remains cases that are handled incorrectly, mostly in JS code now, but it also is probably impossible to make the process entirely safe as long as the minification happens on the templates, given the mix of PHP and HTML / CSS / JS code.

### Contribution checklist (*)
